lib/kubernetes-deploy/deploy_task.rb

```diff
@@ -340,6 +341,10 @@ def deploy_resources(resources, prune: false, verify:, record_summary: true)
       return if resources.empty?
       deploy_started_at = Time.now.utc
 
+      if server_version < Gem::Version.new(@min_version)
+        @logger.warn("Minimum cluster version requirement of #{@min_version} not met.")
+      end
```

Comment:
`deploy_resources` actually gets called several times during the deploy process. Let's check this during the "Initializing deploy" phase. I'm thinking `confirm_cluster_reachable` would be an appropriate point.

That message looks good, but I'd suggest adding something about what the requirement means, i.e. that kubernetes-deploy is not tested against earlier versions of Kubernetes and may not work as expected.

Reply:
Btw one way to 🎩  this would be to start an old version of minikube and run `PRINT_LOGS=1 bundle exec ruby -I test test/integration/kubernetes_deploy_test.rb -n/full_hello/`.

Reply:
Thanks, I was wondering how I could test this stuff!